### PR TITLE
feat(react-query): add `queryOptions`

### LINKED ISSUE #767: [Bug]: Including an object of type UseQueryOptions with select in the queries of useSuspenseQueries results in a type error.
### Package Scope

@suspensive/react-query

### Bug description

Including an object of type UseQueryOptions with select in the queries of useSuspenseQueries results in a type error.
image

For your reference, the type of useGetStringQueryObject is as follows in the picture below.
image

### Expected behavior

There should not be a type error, and the data should be of type string.

### To Reproduce
**tsc --v: Version 4.9.5**

Please refer to the code below :)
```ts
type QueryOptions<T> = UseQueryOptions<Some, unknown, T>;

type Some = {
  id: string;
  createdAt: Date;
};

function useGetSomeQueryObject<T = Some>(
  options?: QueryOptions<T>
): Omit<QueryOptions<T>, "enabled"> {
  return {
    staleTime: Infinity,
    ...options,
    queryKey: ["a", "b"],
    queryFn: () =>
      Promise.resolve({
        id: `${new Date().valueOf()}`,
        createdAt: new Date(),
      }),
  };
}

function useGetStringQueryObject() {
  return useGetSomeQueryObject({
    select: (data) => data.id,
  });
}

const [{ data: string1 }] = useSuspenseQueries({
  queries: [useGetStringQueryObject()], // Type error occurred!
});
```

### Possible Solution

Modifying `select: (data....` to `select?: (data...` in the code below seems to solve the issue.
https://github.com/suspensive/react/blob/08bb5f25f8cf17e61f94d492b505b11d38fe7db0/packages/react-query/src/useSuspenseQueries.ts#L32-L39
It appears that TanStack's React Query has made a similar adjustment. [Link](https://github.com/TanStack/query/pull/6454)

### etc.

_No response_

### PR DESCRIPTION
**Pull request by manudeli**
fix #767 with @wes5510

@wes5510 I guess that this issue is simmilar issue with https://github.com/TanStack/query/issues/6556
I want to solve this issue like https://github.com/TanStack/query/issues/6556#issuecomment-1863164001 (Is this solution same with your opinion?). In my opinion, I think it would be a good idea to add queryOptions

Apr-03-2024 01-54-47

### select type when I hover on field name
image
### select type when I hover on select's arg
I expected TData but it's any!!??🥲
image

I mention you because you suggested this interface(queryOptions) and I've wanted to add it for a while too. I would like to request a review to you with our maintainers.

# Solution: new `queryOptions`

This new queryOptions will solve select's type issue like https://github.com/TanStack/query/issues/6556#issuecomment-1863164001

## More advantages

Tkdodo, The maintainer of TanStack Query explains well why this interface is needed in [video explaining queryOptions in TanStack Query v5](https://youtu.be/bhE3wuB_TuA?feature=shared&t=1697).
You can also use queryOptions in TanStack Query v4.

1. QueryKey management becomes easier by processing queryKey and queryFn together.
2. You can remove unnecessary custom query hooks. This is because they can all be used directly in `useQuery`, `useQueries` of TanStack Query v4, and `useSuspenseQuery`, `useSuspenseQueries`, and `SuspenseQuery` of Suspensive React Query.
3. TanStack Query v5 already supports queryOptions. This Suspensive React Query's `queryOptions` will make migration from TanStack Query v4 to TanStack Query v5 easier.

```jsx /queryOptions/
import { queryOptions, useSuspenseQuery, useSuspenseQueries, SuspenseQuery } from '@suspensive/react-query'
import { useQuery, useQueries, useQueryClient } from '@tanstack/react-query'

const postQueryOptions = (postId) =>
   queryOptions({
     queryKey: ['posts', postId] as const,
     queryFn: ({
       queryKey: [, postId], // You can use queryKey.
     }) => fetch(`https://example.com/posts/${postId}`),
   })

// No need to create custom query hooks.
// You can use queryOptions directly in useQuery, useQueries, useSuspenseQuery, useSuspenseQueries, SuspenseQuery.
const post1Query = useQuery(postQueryOptions(1))
const { data: post1 } = useSuspenseQuery({
   ...postQueryOptions(1);
   refetchInterval: 2000, // Extensibility is clearly expressed in usage.
})
const [post1Query, post2Query] = useQueries({
   queries: [postQueryOptions(1), { ...postQueryOptions(2), refetchInterval: 2000 }],
})
const [{ data: post1 }, { data: post2 }] = useSuspenseQueries({
   queries: [postQueryOptions(1), { ...postQueryOptions(2), refetchInterval: 2000 }],
})
const Example = () => <SuspenseQuery {...postQueryOptions(1)}>{({ data: post1 }) => <>{post1.text}</>}</SuspenseQuery>

// You can easily use queryKey and queryFn in queryClient's methods.
const queryClient = useQueryClient()
queryClient.refetchQueries(postQueryOptions(1))
queryClient.prefetchQuery(postQueryOptions(1))
queryClient.invalidateQueries(postQueryOptions(1))
queryClient.fetchQuery(postQueryOptions(1))
queryClient.resetQueries(postQueryOptions(1))
queryClient.cancelQueries(postQueryOptions(1))

```

## PR Checklist

- [x] I did below actions if need

1. I read the [Contributing Guide](https://github.com/suspensive/react/blob/main/CONTRIBUTING.md)
2. I added documents and tests.

### CODE REVIEW

**Review by minsoo-web — COMMENTED** [review]
Lets Go!!! 🔥🔥

File: docs/suspensive.org/src/pages/docs/react-query/migrate-to-v2.ko.mdx [hunk 1]
```diff
@@ -96,6 +96,53 @@ const PostsPage = ({ authorId }) => (
...
+  ...postQueryOptions(1),
+  refetchInterval: 2000, // 사용처에서 확장성을 명확히 표현됩니다.
+})
+const [post1Query, post2Query] = useQueries({
+  queries: [postQueryOptions(1), { ...postQueryOptions(2), refetchInterval: 2000 }],
+})
+const [{ data: post1 }, { data: post2 }] = useSuspenseQueries({
+  queries: [postQueryOptions(1), { ...postQueryOptions(2), refetchInterval: 2000 }],
+})
+const Example = () => <SuspenseQuery {...postQueryOptions(1)}>{({ data: post1 }) => <>{post1.text}</>}</SuspenseQuery>
```
[the remark this answers is not in the document]
Comment (manudeli) [reply docs/suspensive.org/src/pages/docs/react-query/migrate-to-v2.ko.mdx:133]: Much better!

**Comment by manudeli** [comment]
@wes5510 FYI
If you okay, approve this Pull Request please

File: docs/suspensive.org/src/pages/docs/react-query/migrate-to-v2.ko.mdx [hunk 2]
```diff
@@ -96,6 +96,53 @@ const PostsPage = ({ authorId }) => (
...
+
+```jsx /queryOptions/
+import { queryOptions, useSuspenseQuery, useSuspenseQueries, SuspenseQuery } from '@suspensive/react-query'
+import { useQuery, useQueries, useQueryClient } from '@tanstack/react-query'
+
+const postQueryOptions = (postId) =>
+  queryOptions({
+    queryKey: ['posts', postId] as const,
+    queryFn: ({
+      queryKey: [, postId], // queryKey를 활용할 수 있습니다.
```
[the remark this answers is not in the document]
Comment (manudeli) [reply docs/suspensive.org/src/pages/docs/react-query/migrate-to-v2.ko.mdx:116]: This is just destructuring. check it out again and resolve this conversation please if you okay

File: docs/suspensive.org/src/pages/docs/react-query/migrate-to-v2.ko.mdx [hunk 3]
```diff
@@ -96,6 +96,53 @@ const PostsPage = ({ authorId }) => (
...
+    }) => fetch(`https://example.com/posts/${postId}`),
+  })
+
+// 커스텀 쿼리 훅을 만들 필요가 없습니다.
+// useQuery, useQueries, useSuspenseQuery, useSuspenseQueries, SuspenseQuery에서 직접 queryOptions를 활용할 수 있습니다.
```
[the remark this answers is not in the document]
Comment (manudeli) [reply docs/suspensive.org/src/pages/docs/react-query/migrate-to-v2.ko.mdx:121]: I think it's not major review for feature. so I'd like to get your next Pull Request to express this exactly. Could you do it please?

File: docs/suspensive.org/src/pages/docs/react-query/migrate-to-v2.ko.mdx [hunk 4]
```diff
@@ -96,6 +96,53 @@ const PostsPage = ({ authorId }) => (
...
+
+const postQueryOptions = (postId) =>
+  queryOptions({
+    queryKey: ['posts', postId] as const,
+    queryFn: ({
+      queryKey: [, postId], // queryKey를 활용할 수 있습니다.
+    }) => fetch(`https://example.com/posts/${postId}`),
+  })
+
+// 커스텀 쿼리 훅을 만들 필요가 없습니다.
```
[the remark this answers is not in the document]
Comment (manudeli) [reply docs/suspensive.org/src/pages/docs/react-query/migrate-to-v2.ko.mdx:120]: Same with https://github.com/suspensive/react/pull/828#discussion_r1550907253

**Review by minsoo-web — APPROVED** [review]
I'll make a separate PR and proceed with the translation improvement work!

File: .changeset/chatty-windows-carry.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+"@suspensive/react-query": minor
+---
+
+feat(react-query): add queryOptions
```
Comment (manudeli) [inline .changeset/chatty-windows-carry.md:5]:
```suggestion
feat(react-query): add new queryOptions
```

**Comment by manudeli** [comment]
@wes5510 This is Pull Request for @suspensive/react-query beta version. so I'll merge first. and then we will improve it gradually if need. Your interest has made our library better. thank you!

**Pull request merged**